lib/spack/spack/cmd/test.py

```diff
@@ -101,26 +131,119 @@ def test(parser, args):
-                        time=now,
+                        name=test_name,
                         remove_directory=not args.keep_stage,
                         dirty=args.dirty)
+                    with open(_get_results_file(test_name), 'a') as f:
+                        f.write("%s PASSED\n" % spec.format("{name}-{hash:7}"))
                 except BaseException:
-                    pass  # Test is logged, go on to other tests
+                    with open(_get_results_file(test_name), 'a') as f:
+                        f.write("%s FAILED\n" % spec.format("{name}-{hash:7}"))
         else:
             raise NotImplementedError
+
+
+def test_list(args):
+    """
+    List tests that are running or have available results.
+    """
+    stage_dir = _get_stage()
+    tests = os.listdir(stage_dir)
+
+    # Filter tests by filter argument
+    if args.filter:
+        def create_filter(f):
+            raw = fnmatch.translate('f' if '*' in f or '?' in f
+                                    else '*' + f + '*')
+            return re.compile(raw, flags=re.IGNORECASE)
+        filters = [create_filter(f) for f in args.filter]
+
+        def match(t, f):
+            return f.match(t)
+        tests = [t for t in tests if any(match(t, f) for f in filters)]
+
+    if tests:
+        # TODO: Make these specify results vs active
+        msg = "Spack test results available for the following tests:\n"
+        msg += "        %s\n" % ' '.join(tests)
+        msg += "    Run `spack clean -t` to remove all tests"
+        tty.msg(msg)
+    else:
+        msg = "No test results match the query\n"
+        msg += "        Tests may have been removed using `spack clean -t`"
+        tty.msg(msg)
+
+
+def test_status(args):
+    """
+    Get the current status for a particular Spack test ensemble.
+    """
+    name = args.name
+    stage = _get_stage(name)
+
+    if os.path.exists(stage):
+        # TODO: Make this handle capability tests too
+        tty.msg("Test %s completed" % name)
+    else:
+        tty.msg("Test %s no longer available" % name)
+
+
+def test_results(args):
+    """
+    Get the results for a particular Spack test ensemble.
+    """
+    name = args.name
+    stage = _get_stage(name)
+
+    # TODO: Make this handle capability tests too
+    # The results file may turn out to be a placeholder for future work
+    if os.path.exists(stage):
+        results_file = _get_results_file(name)
+        if os.path.exists(results_file):
+            msg = "Results for test %s: \n" % name
+            with open(results_file, 'r') as f:
+                lines = f.readlines()
+            for line in lines:
+                msg += "        %s" % line
+            tty.msg(msg)
+        else:
+            msg = "Test %s has no results.\n" % name
+            msg += "        Check if it is active with "
+            msg += "`spack test status %s`" % name
+            tty.msg(msg)
+    else:
+        tty.msg("No test %s found in test stage" % name)
```

Comment:
I am getting this message for `libsigsegv` and `openmpi` test results that still exist in `$HOME/.spack/test`.

The command implies, IMHO, that I should be able to see results for completed tests.

Reply:
I did not realize the onus for naming tests for subsequent `status` and `results` queries is on the user.  Now it makes sense.